docs: Add warning on stateful services and shared memory model (#1951)
php/frankenphp: docs/worker.md

Boot your application once and keep it in memory.
FrankenPHP will handle incoming requests in a few milliseconds.

## Warning about stateful design

Unlike with the traditional PHP-FPM model, the application remains loaded in memory between requests. Consequently, any state stored in your services (object properties, singletons, etc.) will be preserved and shared across successive requests handled by the same worker. This can lead to data leaks or inconsistent states if your application is not designed for it.

> The following article summarizes this issue very well and explains how to fix it, notably for Symfony applications using ResetInterface to ensure your services are "clean" for every new request.

Reviewer: ```suggestion
The following article summarizes this issue and explains how to fix it, notably for Symfony applications using ResetInterface to ensure your stateful services are "clean" for every new request.
```

Boot your application once and keep it in memory.
FrankenPHP will handle incoming requests in a few milliseconds.

## A Paradigm Shift

> Warning: This mode introduces a new development paradigm. If your code is not stateless, enabling worker mode risks introducing unexpected behavior.

Reviewer: Thinking about it, I also don't think it introduces a new development paradigm. The behaviour is also not undefined.

> Warning: This mode introduces a new development paradigm. If your code is not stateless, enabling worker mode risks introducing unexpected behavior.

Reviewer: ```suggestion
## Warning about stateful <something>
```

Reviewer: "Warning about stateful design"
or
"persisting state" is good!

Author: done 👍

> Warning: This mode introduces a new development paradigm. If your code is not stateless, enabling worker mode risks introducing unexpected behavior.

Author: I agree, it may have been a little too alarmist. 😄

Unlike with the traditional PHP-FPM model, the application remains loaded in memory between requests. Consequently, any state stored in your services (object properties, singletons, etc.) will be preserved and shared across successive requests handled by the same worker. This can lead to data leaks or inconsistent states if your application is not designed for it.
The following article summarizes this issue very well and explains how to fix it, notably for Symfony applications using ResetInterface to ensure your services are "clean" for every new request.

**Additional Resources:**

> - Article: [Getting your Symfony app ready for Swoole, RoadRunner, and FrankenPHP](https://dev.to/sergiid/getting-symfony-app-ready-for-swoole-roadrunner-and-frankenphp-no-ai-involved-2d0g) - Explains the problem in detail and presents solutions.

Reviewer: ```suggestion
- Article: [Getting your Symfony app ready for Swoole, RoadRunner, and FrankenPHP](https://dev.to/sergiid/getting-symfony-app-ready-for-swoole-roadrunner-and-frankenphp-no-ai-involved-2d0g)
```

I wouldn't want to label it a "problem" again.
